### PR TITLE
Issue 9738 tests

### PR DESCRIPTION
**Pull request by dfm88**
@auvipy I've added tests for pr https://github.com/celery/celery/pull/9740.

### Original PR changes
- simple safer access to `parent_task.request`

### Tests Added
- **Unit tests** in `t/unit/tasks/test_tasks.py` 
- **Integration tests** in `t/integration/test_tasks.py`
- Both cover single task execution and nested parent-child relationships (3 levels deep)
- Added edge case testing for parent tasks without root_id

The tests are quite similar between unit and integration - we could avoid the copy-paste with a global conftest, but there isn't one and it's probably not necessary for this simple case.

## Canvas Operations Limitation

However, the current solution won't work for canvas operations (chain, group, chord) like it does with `apply_async`. For example:

```python
# This won't have proper root/parent IDs:
chain(task1.s(), task2.s(), task3.s()).apply()
```

The issue is that canvas classes have their own [apply()](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) methods in [canvas.py](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) that don't establish the proper ID context before calling individual tasks. The fix only handles the basic [task.apply()](vscode-file://vscode-app/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-sandbox/workbench/workbench.html) method, but canvas operations bypass this.

For full canvas support, the methods in `canvas.py::_chain::apply()`, `canvas.py::group::apply()` and `canvas.py::_chord: apply()`, classes would need similar updates to establish the ID hierarchy like their `apply_async` methods do.

### CODE REVIEW

**Comment by auvipy** [comment]
thanks dude

File: t/integration/test_tasks.py [hunk 1]
```diff
@@ -495,6 +495,86 @@ def test_soft_time_limit_exceeding_time_limit(self):
             assert result.status == 'FAILURE'
 
 
+class test_apply_tasks:
+    """Tests for tasks called via apply() method."""
+
+    def test_apply_single_task_ids(self, manager):
```
[the remark this answers is not in the document]
Comment (auvipy) [reply t/integration/test_tasks.py:502]: what do you think about this suggestion?

[the remark this answers is not in the document]
Comment (dfm88) [reply t/integration/test_tasks.py:502]: I quite agree as in the PR description 

> The tests are quite similar between unit and integration - we could avoid the copy-paste with a global conftest, but there isn't one and it's probably not necessary for this simple case.

If I'm not missing something, I see that `/unit/` and `/integration/` tests folders are quite "closed" between each other, each folder has his own `conftest.py`file...but yes, since tests are pretty equals it would be ideal to have a common place where to have these mock tasks...where do you think it would be better to place these mock tasks, in a global `conftest.py` under `/t/` folder or in a simple common function?

[the remark this answers is not in the document]
Comment (dfm88) [reply t/integration/test_tasks.py:502]: or maybe, since `apply()` is sync and tests are actually equals, maybe leave only one between `integration` and `unit` tests

[the remark this answers is not in the document]
Comment (auvipy) [reply t/integration/test_tasks.py:502]: yeah right. lets keep them separate. as they are now

**Comment by auvipy** [comment]
is it possible to improve / increase unit test coverage?

**Comment by dfm88** [comment]
> is it possible to improve / increase unit test coverage?

Ok I'll check the output of `pytest-cov`

**Comment by dfm88** [comment]
> All modified and coverable lines are covered by tests ✅

Actually I don't know how to increase coverage for these specific changes since they seems to bee all covered by new tests

[CodeCov](https://app.codecov.io/gh/celery/celery/pull/9784?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=celery)

**Comment by auvipy** [comment]
oh sorry no need
